HADOOP-16315. ABFS: transform full UPN for named user in AclStatus

DadanielZ opened the pull request:
Fixed identity conversion for AclEntry when calling getAclStatus()

All tests passed my US-west account:
xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 25
Tests run: 190, Failures: 0, Errors: 0, Skipped: 23

non-xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 207
Tests run: 190, Failures: 0, Errors: 0, Skipped: 15

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 29 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1036 | trunk passed |
| +1 | compile | 26 | trunk passed |
| +1 | checkstyle | 18 | trunk passed |
| +1 | mvnsite | 33 | trunk passed |
| +1 | shadedclient | 698 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 19 | trunk passed |
| 0 | spotbugs | 50 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 47 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 25 | the patch passed |
| +1 | compile | 25 | the patch passed |
| +1 | javac | 25 | the patch passed |
| -0 | checkstyle | 13 | hadoop-tools/hadoop-azure: The patch generated 4 new + 2 unchanged - 0 fixed = 6 total (was 2) |
| +1 | mvnsite | 27 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 718 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | the patch passed |
| +1 | findbugs | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 85 | hadoop-azure in the patch passed. |
| +1 | asflicense | 32 | The patch does not generate ASF License warnings. |
| | | 3008 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 2b29718fdd6c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 77170e7 |
| Default Java | 1.8.0_191 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/1/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/1/testReport/ |
| Max. process+thread count | 446 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/1/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

ThomasMarquardt commented on hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/ITestAbfsIdentityTransformer.java:
```diff
@@ -262,6 +266,10 @@ public void transformAclEntriesForSetRequest() throws IOException {
     identityTransformer = getTransformerWithCustomizedIdentityConfig(config);
 
+    // make a copy
+    aclEntries = Lists.newArrayList(aclEntriesToBeTransformed);
```
L270 can be removed since the first copy was not modified.

ThomasMarquardt commented:
+1 LGTM

We will need to make this extensible in the future to support other on-premise to cloud identity mappings.

steveloughran commented on hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/oauth2/IdentityTransformer.java:
```diff
@@ -218,7 +217,62 @@ public String transformUserOrGroupForSetRequest(String userOrGroup) {
+  /**
+   * Perform Identity transformation when calling GetAclStatus()
+   * If the AclEntry type is a user or group, and its name is one of the following:
+   * 1. $superuser:
```
surround with `pre` tags so that in javadocs this comes out formatted.

steveloughran commented:
just seen this. Don't be afraid to ping myself, sean or aaron fabbri when you want stuff reviewed.

@DadanielZ : can you address @ThomasMarquardt's comment and the one I' ve just added about javadocs? Then it's good to go

DadanielZ commented:
@steveloughran @ThomasMarquardt sorry for the late response, I was overwhelmed with other work in the past two weeks. I've added the pre tag and updated the test.
Thanks!

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 32 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1050 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 32 | trunk passed |
| +1 | shadedclient | 658 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | trunk passed |
| 0 | spotbugs | 51 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 50 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 24 | the patch passed |
| +1 | javac | 24 | the patch passed |
| -0 | checkstyle | 14 | hadoop-tools/hadoop-azure: The patch generated 4 new + 2 unchanged - 0 fixed = 6 total (was 2) |
| +1 | mvnsite | 27 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 727 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 19 | the patch passed |
| +1 | findbugs | 57 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 84 | hadoop-azure in the patch passed. |
| +1 | asflicense | 26 | The patch does not generate ASF License warnings. |
| | | 2983 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 170ac3abcb8c 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 1732312 |
| Default Java | 1.8.0_212 |
| checkstyle | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/2/artifact/out/diff-checkstyle-hadoop-tools_hadoop-azure.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/2/testReport/ |
| Max. process+thread count | 448 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/2/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
LGTM. Can you address the checkstyle whitespace complaints & rerun the tests?

DadanielZ commented:
rerun the tests, all passed:
xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 25
Tests run: 190, Failures: 0, Errors: 0, Skipped: 23

non-xns account:
Tests run: 41, Failures: 0, Errors: 0, Skipped: 0
Tests run: 393, Failures: 0, Errors: 0, Skipped: 207
Tests run: 190, Failures: 0, Errors: 0, Skipped: 15

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 38 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1179 | trunk passed |
| +1 | compile | 29 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 35 | trunk passed |
| +1 | shadedclient | 723 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 54 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 52 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 41 | the patch passed |
| +1 | compile | 26 | the patch passed |
| +1 | javac | 26 | the patch passed |
| +1 | checkstyle | 18 | the patch passed |
| +1 | mvnsite | 28 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 737 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 19 | the patch passed |
| +1 | findbugs | 53 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 79 | hadoop-azure in the patch passed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 3218 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.7 Server=18.09.7 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/3/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f3a099cb0b34 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 4a70a0d |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/3/testReport/ |
| Max. process+thread count | 417 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/3/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 34 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1088 | trunk passed |
| +1 | compile | 28 | trunk passed |
| +1 | checkstyle | 19 | trunk passed |
| +1 | mvnsite | 32 | trunk passed |
| +1 | shadedclient | 660 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | trunk passed |
| 0 | spotbugs | 49 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 47 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 25 | the patch passed |
| +1 | compile | 23 | the patch passed |
| +1 | javac | 23 | the patch passed |
| +1 | checkstyle | 13 | the patch passed |
| +1 | mvnsite | 29 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 713 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 52 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 75 | hadoop-azure in the patch passed. |
| +1 | asflicense | 23 | The patch does not generate ASF License warnings. |
| | | 2973 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.8 Server=18.09.8 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/4/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux c5091cecad84 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 6282c02 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/4/testReport/ |
| Max. process+thread count | 414 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/4/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
LGTM: +1

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 82 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 1 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1193 | trunk passed |
| +1 | compile | 31 | trunk passed |
| +1 | checkstyle | 22 | trunk passed |
| +1 | mvnsite | 35 | trunk passed |
| +1 | shadedclient | 799 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 24 | trunk passed |
| 0 | spotbugs | 61 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 60 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 31 | the patch passed |
| +1 | compile | 30 | the patch passed |
| +1 | javac | 30 | the patch passed |
| +1 | checkstyle | 19 | the patch passed |
| +1 | mvnsite | 32 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 866 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 23 | the patch passed |
| +1 | findbugs | 70 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 84 | hadoop-azure in the patch passed. |
| +1 | asflicense | 30 | The patch does not generate ASF License warnings. |
| | | 3522 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/5/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 907ca732d94f 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / c0a0c35 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/5/testReport/ |
| Max. process+thread count | 309 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/5/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 71 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1308 | trunk passed |
| +1 | compile | 38 | trunk passed |
| +1 | checkstyle | 28 | trunk passed |
| +1 | mvnsite | 43 | trunk passed |
| +1 | shadedclient | 902 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 25 | trunk passed |
| 0 | spotbugs | 60 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 57 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 33 | the patch passed |
| +1 | compile | 30 | the patch passed |
| +1 | javac | 30 | the patch passed |
| +1 | checkstyle | 18 | the patch passed |
| +1 | mvnsite | 31 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 874 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 22 | the patch passed |
| +1 | findbugs | 66 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 88 | hadoop-azure in the patch passed. |
| +1 | asflicense | 29 | The patch does not generate ASF License warnings. |
| | | 3748 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.0 Server=19.03.0 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/6/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux ea62227ed7f0 4.15.0-52-generic #56-Ubuntu SMP Tue Jun 4 22:49:08 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / e20b195 |
| Default Java | 1.8.0_212 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/6/testReport/ |
| Max. process+thread count | 306 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/6/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
@DadanielZ  feel free to merge this when you are ready....

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 45 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1053 | trunk passed |
| +1 | compile | 24 | trunk passed |
| +1 | checkstyle | 17 | trunk passed |
| +1 | mvnsite | 28 | trunk passed |
| +1 | shadedclient | 670 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 21 | trunk passed |
| 0 | spotbugs | 46 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 45 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 26 | the patch passed |
| +1 | compile | 20 | the patch passed |
| +1 | javac | 20 | the patch passed |
| +1 | checkstyle | 13 | the patch passed |
| +1 | mvnsite | 24 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 718 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 19 | the patch passed |
| +1 | findbugs | 68 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 82 | hadoop-azure in the patch passed. |
| +1 | asflicense | 34 | The patch does not generate ASF License warnings. |
| | | 2992 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/7/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux 02e17aab98fb 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 70b4617 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/7/testReport/ |
| Max. process+thread count | 447 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/7/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

hadoop-yetus commented:
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 41 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| +1 | test4tests | 0 | The patch appears to include 1 new or modified test files. |
||| _ trunk Compile Tests _ |
| +1 | mvninstall | 1088 | trunk passed |
| +1 | compile | 30 | trunk passed |
| +1 | checkstyle | 16 | trunk passed |
| +1 | mvnsite | 30 | trunk passed |
| +1 | shadedclient | 671 | branch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 20 | trunk passed |
| 0 | spotbugs | 46 | Used deprecated FindBugs config; considering switching to SpotBugs. |
| +1 | findbugs | 45 | trunk passed |
||| _ Patch Compile Tests _ |
| +1 | mvninstall | 24 | the patch passed |
| +1 | compile | 20 | the patch passed |
| +1 | javac | 20 | the patch passed |
| +1 | checkstyle | 13 | the patch passed |
| +1 | mvnsite | 24 | the patch passed |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 712 | patch has no errors when building and testing our client artifacts. |
| +1 | javadoc | 18 | the patch passed |
| +1 | findbugs | 49 | the patch passed |
||| _ Other Tests _ |
| +1 | unit | 78 | hadoop-azure in the patch passed. |
| +1 | asflicense | 24 | The patch does not generate ASF License warnings. |
| | | 2990 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.1 Server=19.03.1 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-823/8/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/823 |
| Optional Tests | dupname asflicense compile javac javadoc mvninstall mvnsite unit shadedclient findbugs checkstyle |
| uname | Linux f7b379a080fe 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 397a563 |
| Default Java | 1.8.0_222 |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/8/testReport/ |
| Max. process+thread count | 412 (vs. ulimit of 5500) |
| modules | C: hadoop-tools/hadoop-azure U: hadoop-tools/hadoop-azure |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-823/8/console |
| versions | git=2.7.4 maven=3.3.9 findbugs=3.1.0-RC1 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

steveloughran commented:
merged

DadanielZ commented:
@steveloughran  I was traveling in the past two weeks so didn't get the chance to do it. Thanks for the help to review and commit.
